examples/fluids/src/misc.c

```diff
@@ -465,3 +466,136 @@ PetscErrorCode IntArrayP2C(PetscInt num_entries, PetscInt **array_petsc, CeedInt
+    if (!rank) {
+      PetscCall(PetscSortInt(comm_size, gather_buffer));
+      part_owned_dofs[0]             = gather_buffer[0];              // min
+      part_owned_dofs[1]             = gather_buffer[comm_size - 1];  // max
+      part_owned_dofs[2]             = gather_buffer[median_index];   // median
+      PetscReal part_owned_dof_ratio = (PetscReal)part_owned_dofs[1] / (PetscReal)part_owned_dofs[2];
+      PetscCall(PetscPrintf(comm, "    Owned free nodes                   : %" PetscInt_FMT ", %" PetscInt_FMT ", %" PetscInt_FMT ", %f\n",
+                            part_owned_dofs[0] / num_comp_q, part_owned_dofs[1] / num_comp_q, part_owned_dofs[2] / num_comp_q, part_owned_dof_ratio));
+    }
+
+    PetscCallMPI(MPI_Gather(&local_dofs, 1, PetscInt_MPI, gather_buffer, 1, PetscInt_MPI, 0, comm));
+    if (!rank) {
+      PetscCall(PetscSortInt(comm_size, gather_buffer));
+      part_local_dofs[0]             = gather_buffer[0];              // min
+      part_local_dofs[1]             = gather_buffer[comm_size - 1];  // max
+      part_local_dofs[2]             = gather_buffer[median_index];   // median
+      PetscReal part_local_dof_ratio = (PetscReal)part_local_dofs[1] / (PetscReal)part_local_dofs[2];
+      PetscCall(PetscPrintf(comm, "    Local nodes                        : %" PetscInt_FMT ", %" PetscInt_FMT ", %" PetscInt_FMT ", %f\n",
+                            part_local_dofs[0] / num_comp_q, part_local_dofs[1] / num_comp_q, part_local_dofs[2] / num_comp_q, part_local_dof_ratio));
+    }
+
+    PetscInt num_remote_roots_total = 0;
+    {
+      PetscSF            sf;
+      PetscInt           nrranks;
+      const PetscInt    *roffset, *rmine, *rremote;
+      const PetscMPIInt *rranks;
+      PetscCall(DMGetSectionSF(user->dm, &sf));
+      PetscCall(PetscSFGetRootRanks(sf, &nrranks, &rranks, &roffset, &rmine, &rremote));
+      for (PetscInt i = 0; i < nrranks; i++) {
+        if (rranks[i] == rank) continue;  // Ignore same-part global->local transfers
+        num_remote_roots_total += roffset[i + 1] - roffset[i];
+      }
+    }
+    PetscCallMPI(MPI_Gather(&num_remote_roots_total, 1, PetscInt_MPI, gather_buffer, 1, PetscInt_MPI, 0, comm));
+    if (!rank) {
+      PetscCall(PetscSortInt(comm_size, gather_buffer));
+      part_shared_dofs[0]             = gather_buffer[0];              // min
+      part_shared_dofs[1]             = gather_buffer[comm_size - 1];  // max
+      part_shared_dofs[2]             = gather_buffer[median_index];   // median
+      PetscReal part_shared_dof_ratio = (PetscReal)part_shared_dofs[1] / (PetscReal)part_shared_dofs[2];
+      PetscCall(PetscPrintf(comm, "    Shared nodes                       : %" PetscInt_FMT ", %" PetscInt_FMT ", %" PetscInt_FMT ", %f\n",
+                            part_shared_dofs[0] / num_comp_q, part_shared_dofs[1] / num_comp_q, part_shared_dofs[2] / num_comp_q,
+                            part_shared_dof_ratio));
```

Comment:
A typical run on two processes.
```
  Partition:                             (min,max,median,max/median)
    Owned free nodes                   : 41859, 43941, 41859, 1.049738
    Local nodes                        : 46624, 48112, 46624, 1.031915
    Shared nodes                       : 0, 483, 0, inf
```
The `inf` is a little weird, but I guess not a problem.

I think we need better terminology than "Shared nodes" since that's ambiguous as to whether those are unowned halo nodes or nodes that I own and are referenced by another process. Those have similar but not identical information.

Reply:
Fair. I had intended that each local partition could be evenly divided between Owned "free" nodes, Dirichlet nodes, and non-owned nodes. I guess "non-owned" nodes would be more clear than "shared" nodes?

Reply:
Perhaps halo nodes or ghost nodes?

Reply:
I went with Ghost.